lib/rubill/session.rb

```diff
@@ -67,8 +67,19 @@ def _post(url, data, retries=0)
     end
 
     def self._post(url, options)
-      body = post(url, body: options, headers: default_headers).body
-      result = JSON.parse(body, symbolize_names: true)
+      if options.key?(:fileName)
+        file = StringIO.new(options.delete(:content))
+      end
+
+      post_options = {
+        body: options,
+        headers: default_headers,
+      }
+
+      post_options[:file] = file if file
```

Comment:
You can test it yourself by trying this code snippet in your console:

    2.2.1 :002 > options = {}
     => {}
    2.2.1 :003 >       if options.key?(:fileName)
    2.2.1 :004?>           file = StringIO.new(options.delete(:content))
    2.2.1 :005?>         end
     => nil
    2.2.1 :006 >   file
     => nil

Or even simpler:

    if false
        foo = 1
    end

    foo

name defines in block, not in a condition.